ServerCore/Pages/Teams/Signup.cshtml.cs

```diff
@@ -0,0 +1,51 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+using System.Threading.Tasks;
+
+namespace ServerCore.Pages.Teams
+{
+    public class SignupModel : EventSpecificPageModel
+    {
+        public bool IsMicrosoft { get; set; }
+
+        public SignupModel(PuzzleServerContext serverContext, UserManager<IdentityUser> manager) : base(serverContext, manager)
+        {
+        }
+
+        public async Task<IActionResult> OnGetAsync()
+        {
+            if (LoggedInUser == null)
+            {
+                return Challenge();
+            }
+
+            if ((EventRole != EventRole.play && EventRole != EventRole.admin)
+                || IsNotAllowedInInternEvent()
+                || (EventRole == EventRole.admin && !await LoggedInUser.IsAdminForEvent(_context, Event)))
+            {
+                return Forbid();
+            }
+
+            int existingTeamId = await GetTeamId();
+            if (EventRole == EventRole.play && existingTeamId != -1)
+            {
+                return RedirectToPage("/Teams/Details", new { teamId = existingTeamId });
+            }
+
+            if (LoggedInUser.Email.EndsWith("@microsoft.com"))
+            {
+                IsMicrosoft = true;
+            }
+            else if (!string.IsNullOrWhiteSpace(LoggedInUser.EmployeeAlias))
+            {
+                IsMicrosoft = true;
```

Comment:
Oh this is not even close to true. People put all sorts of things in there.

Reply:
I know it's beyond the scope of this PR, but I really don't think we can trust that EmployeeAlias is valid without either doing auth or sending a verification email. I wouldn't use random user-typed garbage for automatic validation of team constraints.

Reply:
That last part *is* within the scope of this PR though.

Reply:
Do we have better validation for this elsewhere that I should use? This doesn't exactly validate anything, it just gives the user access to the free agent system (based on what I think was your idea that if we limit it to FTEs/interns, we don't have to worry about auto teams not having enough FTEs/interns).

Reply:
tldr: I think the else clause at line 43 is going to be abused and cause trouble, and should be removed.

Reply:
Switched to only allowing ```@microsoft.com```. When MaxExternalsPerTeam is the whole team, anyone is allowed to use this